File: sample/src/main/java/me/xiaopan/sketchsample/fragment/GaussianBlurImageProcessorTestFragment.java
package me.xiaopan.sketchsample.fragment;

import android.os.Bundle;
import android.util.DisplayMetrics;
import android.view.View;
import android.widget.SeekBar;
import android.widget.TextView;

import butterknife.BindView;
import me.xiaopan.sketch.display.TransitionImageDisplayer;
import me.xiaopan.sketch.process.GaussianBlurImageProcessor;
import me.xiaopan.sketchsample.AssetImage;
import me.xiaopan.sketchsample.BaseFragment;
import me.xiaopan.sketchsample.BindContentView;
import me.xiaopan.sketchsample.R;
import me.xiaopan.sketchsample.widget.SampleImageView;

@BindContentView(R.layout.fragment_gaussian_blur)
public class GaussianBlurImageProcessorTestFragment extends BaseFragment {
    @BindView(R.id.image_gaussianBlurFragment)
    SampleImageView imageView;

    @BindView(R.id.seekBar_gaussianBlurFragment)
    SeekBar seekBar;

    @BindView(R.id.text_gaussianBlurFragment)
    TextView progressTextView;

    private int progress = 15;

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        // 通过maxSize限制缩小读到内存的图片的尺寸，尺寸越小高斯模糊越快
        DisplayMetrics metrics = getResources().getDisplayMetrics();
        imageView.getOptions().setMaxSize(metrics.widthPixels / 4, metrics.heightPixels / 4);

        imageView.getOptions().setImageDisplayer(new TransitionImageDisplayer());

        seekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                progressTextView.setText(String.format("%d/%d", seekBar.getProgress(), seekBar.getMax()));
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {

            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {
                progress = seekBar.getProgress();
                apply();
            }
        });

        seekBar.setMax(100);
        seekBar.setProgress(progress);

        apply();
    }

    private void apply() {
        imageView.getOptions().setImageProcessor(GaussianBlurImageProcessor.makeRadius(progress));
        imageView.displayImage(AssetImage.MEI_NV);
    }
}

File: sample/src/main/java/me/panpf/sketch/sample/fragment/GaussianBlurImageProcessorTestFragment.kt
package me.panpf.sketch.sample.fragment

import android.os.Bundle
import android.view.View
import android.widget.SeekBar
import kotlinx.android.synthetic.main.fragment_gaussian_blur.*
import me.panpf.sketch.display.TransitionImageDisplayer
import me.panpf.sketch.process.GaussianBlurImageProcessor
import me.panpf.sketch.sample.AssetImage
import me.panpf.sketch.sample.BaseFragment
import me.panpf.sketch.sample.BindContentView
import me.panpf.sketch.sample.R

@BindContentView(R.layout.fragment_gaussian_blur)
class GaussianBlurImageProcessorTestFragment : BaseFragment() {

    private var progress = 15

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        // 通过maxSize限制缩小读到内存的图片的尺寸，尺寸越小高斯模糊越快
        val metrics = resources.displayMetrics
        image_gaussianBlurFragment.options.setMaxSize(metrics.widthPixels / 4, metrics.heightPixels / 4)

        image_gaussianBlurFragment.options.displayer = TransitionImageDisplayer()

        seekBar_gaussianBlurFragment.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                text_gaussianBlurFragment.text = String.format("%d/%d", seekBar.progress, seekBar.max)
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {

            }

            override fun onStopTrackingTouch(seekBar: SeekBar) {
                progress = seekBar.progress
                apply()
            }
        })

        seekBar_gaussianBlurFragment.max = 100
        seekBar_gaussianBlurFragment.progress = progress

        apply()
    }

    private fun apply() {
        image_gaussianBlurFragment.options.processor = GaussianBlurImageProcessor.makeRadius(progress)
        image_gaussianBlurFragment.displayImage(AssetImage.MEI_NV)
    }
}
